src/Libraries/PythonNodeModelsWpf/PythonNode.cs

```diff
@@ -61,13 +93,7 @@ public void CustomizeView(PythonNode nodeModel, NodeView nodeView)
 
             learnMoreItem.Click += OpenPythonLearningMaterial;
 
-            var availableEngineNames = PythonEngineManager.Instance.AvailableEngines.Select(x => x.Name).ToList();
-            // Add the serialized Python Engine even if it is missing (so that the user does not see an empty slot)
-            if (!availableEngineNames.Contains(nodeModel.EngineName))
-            {
-                availableEngineNames.Add(nodeModel.EngineName);
-            }
-            availableEngineNames.ForEach(x => AddPythonEngineToMenuItems(x));
+            PythonNodeUtils.GetEngineNames(nodeModel).ForEach(x => AddPythonEngineToMenuItems(x));
```

Comment:
What if the menu already contains the engine name(s) returned from `GetEngineNames`?

Reply:
done